tests/unittest_manager.py

```diff
@@ -220,6 +220,21 @@ def test_ast_from_module_name_zip(self):
                 os.path.sep.join(["data", os.path.normcase("MyPyPa-0.1.0-py2.5.zip")])
             )
 
+    def test_ast_from_module_name_pyz(self):
+        try:
+            linked_file_name = os.path.join(
+                resources.RESOURCE_PATH, "MyPyPa-0.1.0-py2.5.pyz"
+            )
+            os.symlink(
+                os.path.join(resources.RESOURCE_PATH, "MyPyPa-0.1.0-py2.5.zip"),
+                linked_file_name,
+            )
+
+            with self._restore_package_cache():
+                self._test_ast_from_zip(linked_file_name)
+        finally:
+            os.remove(linked_file_name)
```

Comment:
We should use the ``tmp_path`` fixture from pytest.

Reply:
I think this test is sufficient. A pyz is just a Python package and `__main__.py` in a zip file. See https://bugs.python.org/issue1739468. This was surprisingly easy because astroid already supported zips via astroid.manager.zip_import_data. I just had to expand the file extension filter to allow a pyz through.

tmp_path from pytest doesn't appear to be used in any of the other tests. I tried to follow the style/pattern of other tests. See temporary symlinks created in unittest_modutils.py.

Reply:
It's true the style of other code is not using ``tmp_path`` but it's because it's coming directly from 15 years ago and tmp_path was not available then. You have a point though we should probably refactor the whole code at once.